ui/app/templates/components/raft-storage-restore.hbs

```diff
@@ -0,0 +1,70 @@
+<PageHeader as |p|>
+  <p.top>
+    <nav class="key-value-header breadcrumb">
+      <ul>
+        <li>
+          <span class="sep">&#x0002f;</span>
+          {{#link-to "vault.cluster.storage"}}
+            Raft Storage
+          {{/link-to}}
+        </li>
+      </ul>
+    </nav>
+  </p.top>
+  <p.levelLeft>
+    <h1 class="title is-3">
+      Restore Snapshot
+    </h1>
+  </p.levelLeft>
+</PageHeader>
+
+<form {{action (perform this.restore this.file) on="submit"}}>
+  <MessageError @errors={{this.errors}} />
+
+  {{#if this.isUploading}}
+    <div class="box is-sideless is-fullwidth is-marginless">
+    <AlertBanner
+      @type="warning"
+      @title="Uploading your file..."
+      @message="Raft snapshots can be very large files. Be prepared for uploading the file to take some time."
+      />
+    </div>
+    <div class="box is-fullwidth is-shadowless">
+    <button type="button" class="button" onclick={{action "cancelUpload"}}>
+      Cancel upload
+    </button>
+    </div>
+  {{else}}
+    <div class="box is-sideless is-fullwidth is-marginless">
+      <AlertBanner
+        @type="warning"
+        @title="This might take a while"
+        @message="Raft snapshots can be very large files. Be prepared for uploading the file to take some time."
+        />
+      <FileToArrayBuffer @onChange={{action (mut file)}} />
```

Comment:
the name of this component is a little confusing to me. it seems like it's more of a `FileUpload` or `SnapshotUploadInput` or something similar. i'm not sure what `Array` means in this context -- could you help me understand your thought process?

Reply:
Sure! It's based on the `FileReader` method that the component uses (the `readAs*` methods here: https://developer.mozilla.org/en-US/docs/Web/API/FileReader#Methods) We should probably combine these with our other file-related ones. Currently there's 3 components, they all take in a file and then emit that file in a certain format: one that reads a file and converts to base64 string, one that reads the file as text, and this one that reads a file and raw bytes. The JS data structure for that is an `ArrayBuffer` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer).